### PR TITLE
use feature branch for cadence->flow-go sync PRs

### LINKED ISSUE #1415: Update flow-go CI action to update feature branch
Currently CI automatically opens an update PR in flow-go.

Update it to open PRs against a feature branch instead.

### PR DESCRIPTION
**Pull request by robert-e-davidson3**
Creates feature-secure-cadence branch off of master, if needed.

Closes #1415 

## Description

We want to keep the flow-go and cadence repos synchronized. But that does not always extend to their master branches.

To this end every cadence PR that makes it into master creates a PR in the flow-go repo. They use a version of cadence that corresponds to the cadence PR that made it into master.

Currently we base off of the flow-go master branch. This is a problem because our cadence and flow-go releases are disjoint while cadence is undergoing major breaking changes in preparation for the secure cadence release. We don't want to add such breaking changes to flow-go releases in the meantime.

The solution here is to branch off of master into a new branch `feature-secure-cadence`. This branch needs to be periodically (manually) synchronized with the flow-go master branch.

When it's time to release secure cadence, the feature branch will be merged into master.

______

- [X] Targeted PR against `master` branch
- [X] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [X] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [X] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Comment by robert-e-davidson3** [comment]
QUESTION: How do I test this?
1. Is there an easy way?
2. Do I need to spin up my own set of repos and branches?
3. Or do I just merge to master and clean up any messes bugs create?

**Review by turbolent — APPROVED** [review]
Nice!

**Comment by turbolent** [comment]
@robert-e-davidson3 feel free to merge

File: .github/workflows/sync-flow-go.yml [hunk 1]
```diff
@@ -32,11 +33,12 @@ jobs:
           repository: onflow/flow-go
       
       # get the latest update branch name to base the PR on that branch
+      # else, use the "feature-secure-cadence" branch
       - name: Get onflow/flow-go base branch name
         run: |
           git fetch
           BRANCH=$(git branch -r -l "*auto-cadence-upgrade/*" --format "%(refname:lstrip=3)" | sort -r | head -n 1)
-          [ -z $BRANCH ] && BRANCH=master
+          [ -z $BRANCH ] && BRANCH=feature-secure-cadence
```
Comment (SupunS) [inline .github/workflows/sync-flow-go.yml:41]: Shall we name this branch `feature/secure-cadence`? In `flow-go` repo, branches starts with `feature/***` has some special semantics (e.g: Need reviews for merging, CI is enabled, etc.)
```suggestion
          [ -z $BRANCH ] && BRANCH=feature/secure-cadence
```

File: .github/workflows/sync-flow-go.yml [hunk 2]
```diff
@@ -45,6 +47,16 @@ jobs:
           NEW_BRANCH=auto-cadence-upgrade/$( date +%s )/${{ github.event.pull_request.head.ref }}
           echo "NEW_BRANCH=$NEW_BRANCH" >> $GITHUB_ENV
 
+      # create feature branch if needed
+      - name: Create feature-secure-cadence branch
```
Comment (SupunS) [inline .github/workflows/sync-flow-go.yml:51]: @robert-e-davidson3 Shouldn't this match with the above branch name (`feature/secure-cadence`)?

File: .github/workflows/sync-flow-go.yml [hunk 3]
```diff
@@ -45,6 +47,16 @@ jobs:
...
+        uses: peterjgrainger/action-create-branch@v2.0.1
+        if: ${{ env.BASE_BRANCH == "feature-secure-cadence" }}
```
Comment (SupunS) [inline .github/workflows/sync-flow-go.yml:53]: same here

File: .github/workflows/sync-flow-go.yml [hunk 4]
```diff
@@ -45,6 +47,16 @@ jobs:
...
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          branch: feature-secure-cadence
```
Comment (SupunS) [inline .github/workflows/sync-flow-go.yml:57]: also here

**Pull request merged**